Sandbox: increase selector scope when targeting to iframe element

retrofox opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/master/docs/contributors/repository-management.md#pull-requests. -->

## Description
This PR changes the default CSS code in order to increase the selector scope when targeting the iframe.

## How has this been tested?
It shouldn't affect the current implementations. Just confirm that the Sandbox styles are applied properly after these changes.


## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->

talldan commented on packages/components/src/sandbox/index.js:
```diff
@@ -242,3 +243,5 @@ export default function Sandbox( {
 	);
 }
+
+export default withFilters( 'editor.Sandbox' )( Sandbox );
```
The `editor` part here refers to the package, so this is incorrect.

retrofox replied:
Yes, you're right. I didn't sure about the name neither. We can change it in case we consider this approach valid to go. 👍

talldan commented on packages/components/src/sandbox/README.md:
```diff
@@ -15,3 +15,24 @@ const MySandBox = () => (
+	);
+};
+
+addFilter( 'editor.Sandbox', 'jetpack/loom-custom-sandbox', extendedSandboxed );
```
Ideally we wouldn't put something like 'jetpack' in the examples.

retrofox replied:
sorry was testing in my local env...

talldan reviewed (dismissed):
What's the use case for adding a filter to Sandbox?

I'm not sure I see how this would be useful—the filter would be applied to every use of the Sandbox component, no matter its purpose.

The example mentions embeds, but Sandbox could also be used outside of the embed block.

Seems like it'd be better to be able to add a filter at the place where Sandbox is implemented.

ellatrix commented:
Would love to know the use case as well. I think we should be very conservative with whole component filters and prefer prop filters where possible.

retrofox commented:
> Would love to know the use case as well. 

Yes.

**first**,`<Sandbox />` applies some styles by default. The following is the most important in this example:

```css
body > div > iframe {
	 width: 100%;
}
```

**second**, styles defined above apply `width: 100%` to en embed block, for instance:
<img width="885" alt="Screen Shot 2020-11-10 at 8 38 05 AM" src="https://user-images.githubusercontent.com/77539/98669457-2d479380-2330-11eb-9679-f7371875dee2.png">

**third**, I've registered another oembed provider and implemented its variation,

**fouth**, for X reason, and please let's take it as a condition, when we insert video, it changes the markup, adding a wrapper element to the iFrame, which makes to lose the styles defined by default by the `<Sanddox />` component.

<img width="910" alt="Screen Shot 2020-11-10 at 8 44 22 AM" src="https://user-images.githubusercontent.com/77539/98670383-68969200-2331-11eb-95c4-166e18e57610.png">

My suggestion was to make the component extensible. But I'm totally happy to trying a different approach.

> I think we should be very conservative with whole component filters and prefer prop filters where possible.

Fair enough. Could you develop a little bit more here? Just want to know what's the right criteria. Thanks in advance.

retrofox commented:
> What's the use case for adding a filter to Sandbox?

Commented [here](https://github.com/WordPress/gutenberg/pull/26842#issuecomment-724654785).
 
> I'm not sure I see how this would be useful—the filter would be applied to every use of the Sandbox component, no matter its purpose.

What's the big deal about what is the downside side of this?
 
> The example mentions embeds, but Sandbox could also be used outside of the embed block.

Correct.

> Seems like it'd be better to be able to add a filter at the place where Sandbox is implemented.

Does this idea make sense? If so, I could move the filter to another component.

talldan commented:
> What's the big deal about what is the downside side of this?

When adding a filter like this it's not just about the one use case you might have, but how this might be used generally as it'll become a public WordPress API that's supported indefinitely. I don't see how the current implementation is useful for the majority of users given there's no way to know which `Sandbox` will be affected.

> Does this idea make sense? If so, I could move the filter to another component.

I don't know that you actually need a filter. `Sandbox` already takes a `scripts` prop. In the embed block this is set from the block's `preview` attribute, so I think you can just use an existing filter like `editor.BlockEdit` to modify the attribute:
https://developer.wordpress.org/block-editor/developers/filters/block-filters/#editor-blockedit

ellatrix commented:
Is it possible to update the default styles in Gutenberg to work better with your embed?

retrofox commented:
> When adding a filter like this it's not just about the one use case you might have, but how this might be used generally as it'll become a public WordPress API that's supported indefinitely.

Yeah, thanks for the clarification. 

> I don't see how the current implementation is useful for the majority of users given there's no way to know which Sandbox will be affected.

Yes, got it. :+1:

> In the embed block this is set from the block's `preview` attribute, so I think you can just use an existing filter like `editor.BlockEdit` to modify the attribute:
> https://developer.wordpress.org/block-editor/developers/filters/block-filters/#editor-blockedit

Is it possible to do it when it's a block variation in a consistent way? The _embed_ block stores the variation slug in an attribute, but it isn't part of the API.

retrofox commented:
> Is it possible to update the default styles in Gutenberg to work better with your embed?

Totally doable. But that's what I wanted to avoid this approach. The style change is not a big deal, though, and pretty sure it shouldn't impact the app.

Thanks.

retrofox committed:
sandbox: set iframe selectors more generic
This change will allow to apply the default styles for the iframe more flexible generic way, in the sense it won't require that iframes need to be straigh children of the div container.

retrofox commented:
> Is it possible to update the default styles in Gutenberg to work better with your embed?

[This change](https://github.com/WordPress/gutenberg/pull/26842/commits/28f3d1737f876b98f3391cdf593d233d36509611) should be enough. If you consider doable I can clean up the previous commits, getting rid of the filter approach.

talldan commented:
@retrofox That sounds good. Yep, I say go for it in terms of updating the styles. I'll dismiss my review.

retrofox commented:
Thanks, folks for your help. Rebased / Updated title and description of PR too. 👍

ellatrix reviewed (approved):
Thanks, this is much better :)
